eox_core/edxapp_wrapper/backends/users_h_v1.py

```diff
@@ -64,69 +66,70 @@ def create_edxapp_user(*args, **kwargs):
+    extended_profile_data = request.data.pop("extended_profile_fields", {})
+
+    # Settings to be monkey patched to suit request
+    registration_fields = getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
+    site_configuration = configuration_helpers.get_current_site_configuration()
+    # Backup copies of original configurations
+    _original_settings_features = deepcopy(settings.FEATURES)
+    _original_settings_registration_fields = deepcopy(registration_fields)
+
+    settings.FEATURES['SKIP_EMAIL_VALIDATION'] = not email_validation
+
+    if not comments_service:
+        if 'ENABLE_DISCUSSION_SERVICE' in settings.FEATURES:
+            settings.FEATURES['ENABLE_DISCUSSION_SERVICE'] = False
+
+    if terms_of_service:
+        registration_fields['terms_of_service'] = 'required'
+    else:
+        registration_fields['terms_of_service'] = 'hidden'
 
-    data = {
-        'username': username,
-        'email': email,
-        'password': kwargs.pop("password"),
-        'name': kwargs.pop("fullname"),
-    }
-    # Go ahead and create the new user
-    with transaction.atomic():
-        # In theory is possible to extend the registration form with a custom app
-        # An example form app for this can be found at http://github.com/open-craft/custom-form-app
-        # form = get_registration_extension_form(data=params)
-        # if not form:
-        form = AccountCreationForm(
-            data=data,
-            tos_required=False,
-            # TODO: we need to support the extra profile fields as defined in the django.settings
-            # extra_fields=extra_fields,
-            # extended_profile_fields=extended_profile_fields,
-            # enforce_password_policy=enforce_password_policy,
-        )
-        (user, profile, registration) = do_create_account(form)  # pylint: disable=unused-variable
-
-    site = kwargs.pop("site", False)
-    if site:
-        create_or_set_user_attribute_created_on_site(user, site)
+    if honor_code:
+        registration_fields['honor_code'] = 'required'
     else:
-        errors.append("The user was not assigned to any site")
+        registration_fields['honor_code'] = 'hidden'
 
-    try:
-        create_comments_service_user(user)
-    except Exception:  # pylint: disable=broad-except
-        errors.append("No comments_service_user was created")
-
-    # TODO: link account with third party auth
-
-    lang_pref = kwargs.pop("language_preference", False)
-    if lang_pref:
-        try:
-            preferences_api.set_user_preference(user, LANGUAGE_KEY, lang_pref)
-        except Exception:  # pylint: disable=broad-except
-            errors.append("Could not set lang preference '{} for user '{}'".format(
-                lang_pref,
-                user.username,
-            ))
+    extended_profile_fields = {
+        'extended_profile_fields': {key: '' for key in extended_profile_data}
+        }
+    site_configuration.values.update(extended_profile_fields)
```

Comment:
The same as above. It is not really a good idea to edit the site_configuration values in memory.
On the other hand, why should you modify `extended_profile_fields` on the site. The API should allow you to fill the values on the site, not create new on the flight